src/test/UniswapV3SplitRouteCallee.t.sol

```diff
@@ -0,0 +1,359 @@
+        drip(linkName);
+        (, uint256 ratePost,,,) = vat.ilks(linkName);
+        assertGt(ratePost, ratePre);
+    }
+
+    function barkLink() private returns (uint256 auctionId) {
+        dog.bark(linkName, aliAddr, aliAddr);
+        auctionId = linkClip.kicks();
+    }
+
+    function trimFunctionHash(bytes calldata data) public pure returns (bytes calldata, bytes calldata) {
+        return (data[:4], data[4:]);
+    }
+
+    function buildCalldata(bytes memory signature, bytes memory data) public pure returns (bytes memory) {
+        return abi.encodePacked(signature, data);
+    }
+
+    function reencodedAutoRouterData(bytes memory txData) private view returns (bytes memory output) {
+        (uint256 deadline, bytes[] memory calls) = abi.decode(txData, (uint256, bytes[]));
+        bytes[] memory transformedCalls = new bytes[](calls.length);
+        for (uint256 i = 0; i < calls.length; i++) {
+            (bytes memory signature, bytes memory callWithTrimmedHash) = this.trimFunctionHash(calls[i]);
+            UniswapV3ExactInputParams memory mCalldata = abi.decode(callWithTrimmedHash, (UniswapV3ExactInputParams));
+            UniswapV3ExactInputParams memory modifiedParams = UniswapV3ExactInputParams({
+                path: mCalldata.path,
+                recipient: address(dan),
+                amountIn: mCalldata.amountIn,
+                amountOutMinimum: mCalldata.amountOutMinimum
+            });
+            transformedCalls[i] = buildCalldata(signature, abi.encode(modifiedParams));
+        }
+        output = abi.encode(deadline, transformedCalls);
+    }
+
+    function testBarkLink() public {
+        (,,,, uint256 dustRad) = vat.ilks(linkName);
+        uint256 amountLink = (dustRad / getLinkPriceRay()) * 2;
+        uint256 kicksPre = linkClip.kicks();
+        getLink(amountLink);
+        joinLink(amountLink);
+        frobMax(amountLink, linkName);
+        drip(linkName);
+        uint256 auctionId = barkLink();
+        uint256 kicksPost = linkClip.kicks();
+        assertEq(auctionId, kicksPost);
+        assertEq(kicksPost, kicksPre + 1);
+        (,, uint256 lot, address usr, uint96 tic,) = linkClip.sales(auctionId);
+        assertEq(usr, aliAddr);
+        assertEq(lot, amountLink);
+        assertEq(tic, block.timestamp);
+    }
+
+    function takeLink(uint256 auctionId, uint256 amt, uint256 max, uint256 minProfit, bytes memory txData) public {
+        vat.hope(linkClipAddr);
+        link.approve(UniswapV3Router2, amt);
+        bytes memory data = abi.encode(danAddr, linkJoinAddr, minProfit, address(0), reencodedAutoRouterData(txData));
+        linkClip.take(auctionId, amt, max, danAddr, data);
+    }
+
+    function createLinkAuction()
+        private
+        returns (uint256 auctionId, uint256 amountLinkWad, uint256 auctionPrice, uint256 auctionDebt)
+    {
+        (,,,, uint256 dustRad) = vat.ilks(linkName);
+        amountLinkWad = (dustRad / getLinkPriceRay()) * 2;
```

Comment:
Can you please also fix this (`amountLinkRay` => `amountLinkWad`) in the 1Inch callee test?

Reply:
There are more places in the callees test, for example everywhere we call `createLinkAuction`.
Suggest to search for `amountLinkRay` and validate each occurrence.

Reply:
adjusted